jax/interpreters/ad.py

```diff
@@ -296,13 +300,17 @@ def process_custom_jvp_call(self, _, __, f_jvp, tracers):
     primals_in, tangents_in = unzip2((t.primal, t.tangent) for t in tracers)
     primals_in = map(core.full_lower, primals_in)
     tangents_in = map(instantiate_zeros, tangents_in)
+    tangents_in = [aval.to_zeros() if type(aval) is core.Float0Array
```

Comment:
The reason for adding this here is to fully instantiate the Float0Arrays as regular float zeros because they will be passed into the custom_jvp rules. They are not ignored, like they are on the regular jvp path. The reason for this is not entirely clear to me, and although this work-around makes my tests pass it does not feel like a true fix. Where on the custom_jvp path do we ignore the Zero tangents?

This particular edge case happens when calling a function that internally calls a custom_jvped function with an integer constant, like in nn.softplus here: https://github.com/google/jax/blob/3f8aaabbccb40bb5c1c69e1f15ba7bd30dd3fdbc/jax/nn/functions.py#L41.

Reply:
I think the `aval` name might be misleading here, since the values in `tangents_in` aren't avals but just values. How about

```suggestion
    tangents_in = [t.to_zeros() if type(t) is core.Float0Array else t for t in tangents_in]
```